docs/learn/encyclopedia/contract-development/events.mdx

```diff
@@ -21,13 +21,7 @@ Events are the mechanism that applications off-chain can use to monitor changes
 
 ## How are events emitted?
 
-`ContractEvents` are emitted in Stellar Core's `TransactionMeta`. You can see in the [TransactionMetaV3] XDR below that there is a list of `OperationEvents` called `events`. Each `OperationEvent` corresponds to an operation in a transaction, and itself contains a list of `ContractEvents`. Note that `events` will only be populated if the transaction succeeds. Take a look at the [events guides](../../../build/guides/events/README.mdx) and [this example](../../../build/smart-contracts/example-contracts/events.mdx) to learn more about how to work with events.
-
-:::info
-
-Events are ephemeral, the data retention window is a maximum of 7 days. See the [ingestion guide](../../../build/guides/events/ingest.mdx) to learn how to work with this constraint.
-
-:::
+`ContractEvents` are emitted in Stellar Core's `TransactionMeta`. You can see in the [TransactionMetaV3] XDR below that there is a list of `OperationEvents` called `events`. Each `OperationEvent` corresponds to an operation in a transaction, and itself contains a list of `ContractEvents`. Note that `events` will only be populated if the transaction succeeds.
 
 [transactionmetav3]: #transactionmetav3
 
```

Comment:
Below this there is a line which says:

> An event also contains a data object of any value or type including types defined by contracts using `#[contracttype]`

I don't think it's clear that `#[contracttype]` is referring to a rust macro. Is there a page we can link to which describes how `#[contracttype]` is used in soroban smart contracts?

Reply:
Good idea! Added a cross-reference to the page on [Custom Types](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/types/custom-types)!